### PR TITLE
BHBC-900: Initial project view components

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## Links to jira tickets

https://quartech.atlassian.net/browse/BHBC-900

## This PR contains the following changes

- Moved ProjectPage.tsx to `projects/view` folder
- Updated ProjectPage.tsx to align with initial designs
- Added placeholder components for the details, surveys, attachments content
- Added util for creating formatted date strings
- Added tests

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [x] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Locally + unit tests.

## Screenshots

image

### CODE REVIEW

File: app/src/features/projects/view/ProjectAttachments.tsx [hunk 1]
```diff
@@ -150,58 +120,49 @@ const ProjectPage: React.FC = () => {
...
+        </Paper>
+      </Box>
+      <Box mb={3}>
+        <Paper>
+          <Box p={3}>
+            <Box key={dropzoneInstanceKey}>
+              <DropzoneArea
+                dropzoneText={dropzoneText}
+                filesLimit={10}
+                //fileObjects={files}
```
Comment (sdevalapurkar) [inline app/src/features/projects/view/ProjectAttachments.tsx]: nit: remove?

File: app/src/features/projects/view/ProjectPage.test.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  beforeEach(() => {
+    // clear mocks before each test
+    mockBiohubApi().getProject.mockClear();
+  });
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  it.skip('renders a spinner if no project is loaded', () => {
```
Comment (sdevalapurkar) [inline app/src/features/projects/view/ProjectPage.test.tsx]: nit: remove skip?

**Review by sdevalapurkar — COMMENTED** [review]
looking good :) few small comments

**Review by anissa-agahchen — COMMENTED** [review]
You currently have the file upload under path /surveys, and the surveys under the path /attachments.  I suspect you want to switch those around.

**Comment by anissa-agahchen** [comment]
Also, resolved the code smells ... mostly TODO's, but the file upload was high in cognitive complexity.  I resolved it as 'won't fix' since we're going to be making significant changes to that code in the near future.

**Review by anissa-agahchen — APPROVED** [review]

**Pull request merged**